src/components/encoding-pane/index.tsx

```diff
@@ -162,12 +162,13 @@ class EncodingPanelBase extends React.PureComponent<EncodingPanelProps, {}> {
 
 export const EncodingPane = connect(
   (state: State) => {
+    const presentUndoableState = state.undoable.present;
     return {
-      spec: state.present.shelf.spec,
-      filters: state.present.shelf.spec.filters,
-      schema: state.present.dataset.schema,
+      spec: presentUndoableState.shelf.spec,
```

Comment:
We should use a selector for dataset, shelf, shelfPreview for consistency and for reduce the amount of updates needed if we reshape the state in the future.